File: Ejercicio_2.py
class CuentaBancaria: 
    def __init__(self, saldo_inicial=0):
        self._saldo= saldo_inicial
    
    def depositar(self, cantidad):
        if cantidad > 0: 
            self._saldo += cantidad 
            print(f"Se hizo un deposito de forma exitosa. El saldo actual es de: ${self._saldo}")
        else: 
            print("La cantidad a ingresar no es una cifra valida")

    def retirar(self, cantidad):
        if cantidad > 0 and self._saldo >= cantidad: 
            self._saldo -= cantidad 
            print(f"Se hizo un retiro de forma exitosa. El saldo actual es de: ${self._saldo}")
        else:
            print("La cantidad a retirar no es valida o el saldo es insuficiente")

    def get_saldo(self):
        print(f"Tu saldo actual es de: ${self._saldo}")

mi_cuenta = CuentaBancaria(1000)
mi_cuenta.depositar(500) 
mi_cuenta.retirar(200) 
mi_cuenta.get_saldo()

File: RA 1.1/Ejercicio_2.cpp
#include <iostream>
using namespace std; 

class CuentaBancaria { 
    private: 
        double saldo= 0; 

    public: 
        CuentaBancaria(double saldo_in): saldo(saldo_in) {}
        ~CuentaBancaria() {}

        void depositar(double cantidad) { 
            if (cantidad > 0) { 
                this->saldo += cantidad; 
                cout<<"Se hizo un deposito de forma exitosa. El saldo actual es de: $"<<this->saldo<<endl;
            } else { 
                cout<<"La cantidad a ingresar no es una cifra valida"<<endl; 
            }
        }

        void retirar(double cantidad) { 
            if (cantidad > 0 && this->saldo >= cantidad) { 
                this->saldo -= cantidad; 
                cout<<"Se hizo un retiro de forma exitosa. El saldo actual es de: $"<<this->saldo<<endl;
            } else { 
                cout<<"La cantidad a retirar no es valida o el saldo es insuficiente"<<endl;
            }
        }

        void get_saldo() {
            cout<<"Tu saldo actual es de: $"<<this->saldo<<endl;
        }
};

int main() {
    CuentaBancaria mi_cuenta(1000); 
    mi_cuenta.depositar(500); 
    mi_cuenta.retirar(200); 
    mi_cuenta.get_saldo();

    return 0;
}